airbyte-api/src/main/openapi/config.yaml

```diff
@@ -1595,13 +1595,17 @@ components:
     ConnectionCreate:
       type: object
       required:
+        - connectionName
         - sourceId
         - destinationId
         - status
       properties:
         name:
           type: string
           description: Optional name of the connection
+        connectionName:
```

Comment:
Why not just called it prefix? tableNamePrefix? The connection already has a name. You're adding a field to allow prefixing right? Maybe I'm missing some bigger picture here, but it seems like you could use a really specific clean name here instead of something broad.

Reply:
in the short term, it's a prefix but in one or two weeks, when namespace fields are available, this field becomes the actual namespace name for the destination.

On top of that destination namespace name, you can have an optional suffix namespace whenever the source supports it (in JDBC source connectors) but in all other cases, the suffix will be null.

To conclude, the connectionName can't be empty when used with non-JDBC sources.

Reply:
IMO: This is very complex & we should call things by their names.

if it is a prefix we should call it prefix. 
if it is a default namespace for record which doesn't have a namespace, then we should call it defaultNamespace

ConnectionName seems like a dead field at that point.

Reply:
oops i just realized I called it `namespaceDefault` instead of `defaultNamespace`... I hope that's fine...

Reply:
> in the short term, it's a prefix but in one or two weeks, when namespace fields are available, this field becomes the actual namespace name for the destination.

in that second step, how will this work with not database sources that declare namespaces? if this a default and gets overridden by the namespace set by the source, won't tables in the destination still collide? e.g. if you were pulling from 2 stripe sources both will try to write to `stripe.users`?

Reply:
i'm still a little unsure if this a namespace default or namespace override. I'm a little fuzzy on what's going to happen if this value is not set. it looks like below we are going throw an NPE or end up with tables that are `null_users`. maybe i'm missing something?

Reply:
Normally, the default namespace should always be defined to specify where data will end up in the destination.

The only case where it would be acceptable for it to be `null` is when the stream already has a namespace defined by the source (thus requires a namespace field introduced by #2228) AND when the user wants to mirror exactly the same namespace structure in the destination as in the source (replicate `public.users` from source in `public.users` in destination)

But for the short term, since destinations are still set with one (and it is required), it is not currently mandatory here immediately

Reply:
> in that second step, how will this work with non database sources that declare namespaces?

let's move that thread discussion over there where I will detail more what we do in the next steps.

https://github.com/airbytehq/airbyte/issues/2324